File: tests/files/extract_func_works_after.rb
class F
  def f
    return 'f'
  end
end

def bar(b, c, d, e, f, g, h, k, l, m, j):
  a = a
  a = a + 1
  a+=a
  a++
  print b
  print c + 1
  print d[0]
  print e['e']
  print f.f()
  print g.call()
  if h then end
  while k do end
  until l do end
  loop do break if m end
  [a,b].each do |v| puts "#{a} #{v}" end
  print(j)

  return a
end

def foo(a)
  for j in 1..5 do
    b = 'b'
    c = 'c'
    d = ['d', b]
    e = {'e' => 'e'}
    f = F.new
    g, h = ->() {"g"}, "h"
    k = 'k'
    l = 'l'
    m = 'm'

    a = bar(b, c, d, e, f, g, h, k, l, m, j)

    return a
  end
end

File: tests/files/extract_func_works_after.java
class F {
    public String f() {
        return "f";
    }
}

record E (String e) {}

class Foo {
    private String bar(int a, String b, String c, String[] d, E e, F f, boolean g, boolean h, boolean k, boolean l, String i, int j) {
        a = a + 1;
        a += a;
        a++;
        ++a;
        System.out.println(b);
        System.out.println(c + 1);
        System.out.println(d[0]);
        System.out.println(e.e());
        System.out.println(f.f());
        System.out.println(g);
        if (h) {}
        while (k) {}
        do {} while (l);
        i = "i";
        System.out.println(j);

        return i;
    }

    public String foo(int a) {
        String i;
        for (int j = 0; j < 5; j++) {
            String b = "b";
            String c = "c";
            String[] d = {"d"};
            E e = new E("e");
            F f = new F();
            boolean g = true, h = false;
            boolean k = true;
            boolean l = true;

            var i = bar(a, b, c, d, e, f, g, h, k, l, i, j);

            return i;
        }
        return "";
    }
}
